qemu/tests/block_copy.py

```diff
@@ -298,7 +297,10 @@ def action_when_start(self):
                 bg = utils.InterruptedThread(fun)
                 bg.start()
                 if bg.isAlive():
-                    self.processes.append(bg)
+                    try:
```

Comment:
if your steps is 1, 2 then 3, your configuration should be action_when_start, 1, then 2, 3 in action_after/before_finished 3 in action_before_cleanup